distribution/lib/Standard/Database/0.0.0-dev/src/Connection/Connection.enso

```diff
@@ -70,16 +74,16 @@ type Connection
     ## Returns the name of the current schema.
     schema : Text
     schema self =
-        self.jdbc_connection.with_connection connection->
-            connection.getSchema
+        self.jdbc_connection.with_connection connection->connection.getSchema
 
     ## Returns a new Connection with the specified schema set as default.
 
        Arguments:
         - schema: The name of the schema to connect to.
+        - parent: The parent connection.
     set_schema : Text -> Connection ! Sql_Error
```

Comment:
Because `self` refers to the underlying connection when called through from an implemented version.

So SQLite_Connection for example needs to pass itself in to use the base implementation.

Reply:
Okay, I see.

The design is still confusing to me - Postgres has similar logic but it implements it directly in `Postgres_Connection` without delegating to the underlying `Connection`. Cannot we do the same for `Sqlite_Connection`? Then I guess we can get rid of `Connection.set_schema` altogether, which seems sensible to me as it seems to do nothing else than a check if the schema is actually unchanged - and this check seems more fitting to sit in `Sqlite_Connection`, as its the only "user" of that check (since Postgres has its own separate implementation for that).